File: api/internal/service/CourseService.go
package service

import (
	"smart-elearning/internal/dto/common"
	"smart-elearning/internal/dto/request"
	"smart-elearning/internal/dto/response"
)

type CourseService interface {
	CreateCourse(request *request.CreateCourseRequest, user *common.ClaimUser) (*response.CourseResponse, error)

	GetAllCourse(page int, limit int, user *common.ClaimUser) ([]*response.CourseOverviewResponse, error)

	IsCourseOwner(courseId uint, teacherId uint) (bool, error)
}

File: api-v2/smart-elearning/src/main/java/com/smartelearning/smartelearning/service/CourseService.java
/*
 *  CourseService
 *  @author: minhhieuano
 *  @created 7/10/2025 4:01 PM
 * */


package com.smartelearning.smartelearning.service;

import com.smartelearning.smartelearning.dto.request.CourseRequest;
import com.smartelearning.smartelearning.dto.request.UpdateCourseAvatarRequest;
import com.smartelearning.smartelearning.dto.response.CourseResponse;
import com.smartelearning.smartelearning.entity.Account;

public interface CourseService {
    CourseResponse createCourse(CourseRequest request, Account teacher);

    String updateCourseAvatar(String courseId, UpdateCourseAvatarRequest request, Account teacher);
}
